homeassistant/components/fritz/config_flow.py

```diff
@@ -0,0 +1,244 @@
+        return self.async_show_form(
+            step_id="confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            description_placeholders={"name": self._name},
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        return await self.async_step_start_config()
+
+    async def async_step_start_config(self, user_input=None):
+        """Handle a flow start config."""
+        if user_input is None:
+            return self._show_setup_form_init()
+
+        errors = {}
+
+        self._host = user_input.get(CONF_HOST, DEFAULT_HOST)
+        self._port = user_input.get(CONF_PORT, DEFAULT_PORT)
+        self._username = user_input.get(CONF_USERNAME)
+        self._password = user_input.get(CONF_PASSWORD)
+
+        error = await self.fritz_tools_init()
+
+        if not error:
+            self._name = self.fritz_tools.device_info["model"]
+
+            for entry in self._async_current_entries(include_ignore=False):
+                if entry.data[CONF_HOST] == self._host:
+                    error = "already_configured"
+
+        if error:
+            errors["base"] = error
+            return self._show_setup_form_init(errors)
+
+        return self._async_create_entry()
+
+    async def async_step_reauth(self, data):
+        """Handle flow upon an API authentication error."""
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        self._host = data.data.get(CONF_HOST, DEFAULT_HOST)
+        self._port = data.data.get(CONF_PORT, DEFAULT_PORT)
+        self._username = data.data.get(CONF_USERNAME)
+        self._password = data.data.get(CONF_PASSWORD)
+        return await self.async_step_reauth_confirm()
+
+    def _show_setup_form_reauth_confirm(self, user_input, errors=None):
+        """Show the reauth form to the user."""
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME)
+                    ): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            description_placeholders={"host": self._host},
+            errors=errors or {},
+        )
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self._show_setup_form_reauth_confirm(
+                user_input={CONF_USERNAME: self._username}
+            )
+
+        errors = {}
+
+        self._username = user_input.get(CONF_USERNAME)
+        self._password = user_input.get(CONF_PASSWORD)
+
```

Comment:
These should be required as well so you shouldn't need to use `.get`

Reply:
Done